Lecture 21-22-23/bst.cpp

```diff
@@ -1,205 +1,232 @@
 #include <iostream>
 #include <queue>
-#include <cmath>	
+#include <cmath>
 #include <climits>
 
 using namespace std;
 
-class node{
+class node {
 public:
 	int data;
 	node* left;
 	node* right;
 
-	node(int data){
+	node(int data) {
 		this->data = data;
 		this->left = NULL;
-		this->right =NULL;
+		this->right = NULL;
 	}
 };
 
-node* construct(node* root,int data){
-	if(root==NULL){
+node* construct(node* root, int data) {
+	if (root == NULL) {
 		node* root = new node(data);
 		return root;
 	}
 
-	if(data>root->data){
-		root->right = construct(root->right,data);
-	}else{
-		root->left = construct(root->left,data);
+	if (data > root->data) {
+		root->right = construct(root->right, data);
+	} else {
+		root->left = construct(root->left, data);
 	}
 
 	return root;
 }
 
-node* buildTree(){
+node* constructIteratively(struct node* root, int data) {
+
+	struct node *p == root, *q = new node(data);
```

Comment:
I don't we are using struct in our code, please modify everywhere according to class node not struct node @watashi-wa-sidhant